### PR TITLE
chore(core): decrease default append page size for WAL writers

### PR DESCRIPTION
**Pull request by puzpuzpuz**
Introduces a separate append page size prop (`cairo.wal.writer.data.append.page.size`) for WAL writers. The new prop is set to 1MB instead of 16MB to play better with the default `cairo.wal.segment.rollover.row.count` value which is 200K (that's 1.6MB for a `long` column).

Since multiple WAL writers may co-exist, having 16MB as the default append page size makes the chances of seeing no disk space errors much more probable.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 6 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|

File: pkg/ami/marketplace/assets/server.conf [hunk 1]
```diff
@@ -594,6 +594,41 @@ pg.user=admin
...
+
+#Row count of how many rows are written to the same WAL segment before starting a new segment.
+#cairo.wal.segment.rollover.row.count=200000
+
+# mmap sliding page size that WalWriter uses to append data for each column
+#cairo.wal.writer.data.append.page.size=1M
+
+#Maximum row count that can be squashed together from multiple transactions before applying to the table.
+#Can delay the data visibility under heavy load but improves the throughput. Setting it very low value can have negative effect and delay the data visibility.
+#cairo.wal.commit.squash.row.limit=500000
```
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply pkg/ami/marketplace/assets/server.conf:629]: Thanks for the heads-up. Merged and fixed.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|

File: pkg/ami/marketplace/assets/server.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (ideoma) [reply pkg/ami/marketplace/assets/server.conf:629]: It is updated in one place but here it's still wrong, the property 

```
cairo.wal.commit.squash.row.limit
```

is gone, should only be

```
# Multiplier to cairo.max.uncommitted.rows to calculate the limit of rows that can kept invisible when writing
# to WAL table under heavy load, when multiple transactions are to be applied.
# It is used to reduce the number Out Of Order commits when Out Of Order commits are unavoidable by squashing multiple commits together.
# Setting it very low can increase Out Of Order commit frequency and decrease the throughput.
# Setting it too high may cause excessive memory usage and increase the latency.
#cairo.wal.squash.uncommitted.rows.multiplier=20.0
```

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|

File: pkg/ami/marketplace/assets/server.conf (hunk 1 above)
[the remark this answers is not in the document]
Comment (puzpuzpuz) [reply pkg/ami/marketplace/assets/server.conf]: Yes, I overlooked this place. Thanks for spotting it. Fixed in 9dd3ab1eba834addf338f8195d6daf62ab5e70e8

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|

**Pull request merged**